gui/tooltips.lua

```diff
@@ -0,0 +1,281 @@
+-- Show tooltips on units and/or mouse
+
+local RELOAD = false -- set to true when actively working on this script
+
+local gui = require('gui')
+local widgets = require('gui.widgets')
+local ResizingPanel = require('gui.widgets.containers.resizing_panel')
+
+--------------------------------------------------------------------------------
+
+local follow_units = true;
+local follow_mouse = true;
+local function change_follow_units(new, old)
+    follow_units = new
+end
+local function change_follow_mouse(new, old)
+    follow_mouse = new
+end
+
+local shortenings = {
+    ["Store item in stockpile"] = "Store item",
+}
+
+--------------------------------------------------------------------------------
+
+local TITLE = "Tooltips"
+
+if RELOAD then TooltipControlWindow = nil end
+TooltipControlWindow = defclass(TooltipControlWindow, widgets.Window)
+TooltipControlWindow.ATTRS {
+    frame_title=TITLE,
+    frame_inset=0,
+    resizable=false,
+    frame = {
+        w = 25,
+        h = 4,
+        -- just under the minimap:
+        r = 2,
+        t = 18,
+    },
+}
+
+function TooltipControlWindow:init()
+    self:addviews{
+        widgets.ToggleHotkeyLabel{
+            view_id = 'btn_follow_units',
+            frame={t=0, h=1},
+            label="Follow units",
+            key='CUSTOM_ALT_U',
+            on_change=change_follow_units,
+        },
+        widgets.ToggleHotkeyLabel{
+            view_id = 'btn_follow_mouse',
+            frame={t=1, h=1},
+            label="Follow mouse",
+            key='CUSTOM_ALT_M',
+            on_change=change_follow_mouse,
+        },
+    }
+end
+
+local function GetUnitJob(unit)
+    local job = unit.job
+    if job and job.current_job then
+        return dfhack.job.getName(job.current_job)
+    end
+    return nil
+end
+
+local function GetUnitNameAndJob(unit)
+    local sb = {}
+    sb[#sb+1] = dfhack.units.getReadableName(unit)
+    local jobName = GetUnitJob(unit)
+    if jobName then
+        sb[#sb+1] = ": "
+        sb[#sb+1] = jobName
+    end
+    return table.concat(sb)
+end
+
+local function GetTooltipText(x,y,z)
+    local txt = {}
+    local units = dfhack.units.getUnitsInBox(x,y,z,x,y,z) or {} -- todo: maybe (optionally) use filter parameter here?
```

Comment:
can take a `pos` param to this function and pass `pos, pos` to `getUnitsInBox` -- it supports `pos` params as of https://github.com/DFHack/dfhack/pull/5146